### PR TITLE
[Enhancement][FlatJson] support flat_json_meta function

### PR DESCRIPTION
**Pull request by Seaven**
## What I'm doing:
to read flat json meta from storage

```
MySQL x13>  select flat_json_meta(j1), count(1) from js4[_META_];
+----------------------------------------------------------------------+----------+
| flat_json_meta(j1)                                                   | count(1) |
+----------------------------------------------------------------------+----------+
| ["nulls(TINYINT)","e(JSON)","c(JSON)","b(JSON)","d(JSON)","a(JSON)"] | 1        |
+----------------------------------------------------------------------+----------+
1 row in set
Time: 0.041s
MySQL x13>
```

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java [hunk 1]
```diff
@@ -437,7 +437,7 @@ public void testMetaScanWithCount() throws Exception {
                 "args nullable: false; result nullable: true], " +
                 "max[([max_t1a, VARCHAR, false]); args: VARCHAR; result: VARCHAR; " +
                 "args nullable: false; result nullable: true], " +
-                "dict_merge[([dict_merge_t1a, VARCHAR, false]); args: INVALID_TYPE; " +
+                "dict_merge[([dict_merge_t1a, ARRAY<VARCHAR>, false]); args: INVALID_TYPE; " +
```
Comment (silverbullet233) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java:440]: why intermediate result type changed? Is this safe? Will there be any compatibility issues during grayscale upgrade?

Comment (Seaven) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java:440]: good question, I modify here
```
                slotDescriptor.setType(entry.getKey().getType());
```

Comment (Seaven) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java:440]: I modify input type, not intermediate type. In actually, we use `array<varchar>` now, not `varchar`, I don't know why doesn't crash now....

Comment (silverbullet233) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/ScanTest.java:440]: sorry, I misread, I thought this was the type of intermediate result before…

but no matter what, it seems that it is correct after your modification?

**Review by silverbullet233 — APPROVED** [review]

**Pull request merged**